addons/l10n_cl/models/account_move.py

```diff
@@ -27,23 +26,93 @@ def _get_l10n_latam_documents_domain(self):
-                if self.partner_id.l10n_cl_sii_taxpayer_type == '2':
-                    domain += [('code', '=', '71')]
-                return domain
-            document_type_ids = self.journal_id.l10n_cl_sequence_ids.mapped('l10n_latam_document_type_id').ids
-            domain += [('id', 'in', document_type_ids)]
-            if self.partner_id.l10n_cl_sii_taxpayer_type == '3':
-                domain += [('code', 'in', ['35', '38', '39', '41', '56', '61'])]
+            if self.journal_id.type == 'sale':
+                document_type_ids = self.journal_id.l10n_cl_sequence_ids.mapped('l10n_latam_document_type_id').ids
+            else:  # self.journal_id.type == 'purchase':
+                partner_domain = [
+                    ('country_id.code', '=', 'CL'),
+                    ('internal_type', 'in', ['invoice', 'debit_note', 'credit_note', 'invoice_in'])]
+                if len(self.partner_id) == 0:
+                    pass
+                elif self.partner_id.l10n_cl_sii_taxpayer_type == '1' and self.partner_id_vat != '60805000-0':
+                    partner_domain += [('code', 'not in', ['39', '70', '71', '914', '911'])]
+                elif self.partner_id.l10n_cl_sii_taxpayer_type == '1' and self.partner_id_vat == '60805000-0':
+                    partner_domain += [('code', 'not in', ['39', '70', '71'])]
+                elif self.partner_id.l10n_cl_sii_taxpayer_type == '2':
+                    partner_domain += [('code', 'in', ['70', '71', '56', '61'])]
+                elif self.partner_id.l10n_cl_sii_taxpayer_type == '3':
+                    partner_domain += [('code', 'in', ['35', '38', '39', '41', '56', '61'])]
+                elif not self.partner_id.l10n_cl_sii_taxpayer_type or self.partner_id.country_id != self.env.ref(
+                        'base.cl') or self.partner_id.l10n_cl_sii_taxpayer_type == '4':
+                    partner_domain += [('code', 'in', [])]
+                document_type_obj = self.env['l10n_latam.document.type'].search(partner_domain)
+                document_type_ids = document_type_obj.ids
+            domain = [('id', 'in', document_type_ids)]
         return domain
 
-    @api.constrains('type', 'l10n_latam_document_type_id')
-    def _check_invoice_type_document_type(self):
-        super()._check_invoice_type_document_type()
-        for rec in self.filtered(lambda r: r.company_id.country_id == self.env.ref(
-                'base.cl') and r.l10n_latam_document_type_id):
+    @api.depends('journal_id', 'partner_id', 'company_id', 'type')
+    def _compute_l10n_latam_available_document_types(self):
+        for record in self:
+            if record.journal_id.l10n_latam_use_documents and \
+                    record.journal_id.company_id.country_id == self.env.ref('base.cl'):
+                record.l10n_latam_available_document_type_ids = False
+                for rec in self.filtered(lambda x: x.journal_id and x.l10n_latam_use_documents):
```

Comment:
I think the filtered and part of the if are redundant

Reply:
Probably a bit too many for if for indeed.

Reply:
@nhomar you're right!

Reply:
fixed at 74e4fdf13b77b211273dc83d8674d3fa88e2ccd2

Reply:
finally, removed this method at 78e42bbe4d04c519f84d2dedb5668c8979edb05a